proxy-attestation-server/src/attestation.rs

```diff
@@ -252,18 +257,57 @@ fn convert_csr_to_certificate(csr_der: &[u8]) -> Result<openssl::x509::X509, Pro
 
     // setting the basic constraints extension to 'critical' - meaning required,
     // to 'ca' meaning the certificate is a CA certificate
-    let constraints_extension = openssl::x509::extension::BasicConstraints::new().critical().ca().pathlen(1).build()
+    let mut constraints_extension = openssl::x509::extension::BasicConstraints::new();
+    constraints_extension.critical();
+    constraints_extension.pathlen(1);
+    if is_ca {
+        constraints_extension.ca();
+    }
+    let built_constraints_extension = constraints_extension.build()
         .map_err(|err| {
             println!("proxy-attestation-server::attestation::convert_csr_to_certificate BasicConstraints::new failed:{:?}", err);
             err
         })?;
-    cert_builder.append_extension(constraints_extension)
+    cert_builder.append_extension(built_constraints_extension)
         .map_err(|err| {
             println!("proxy-attestation-server::attestation::convert_csr_to_certificate append_extension failed:{:?}", err);
             err
         })?;
 
-    let private_key = get_ca_key()
+    // Add our custom extension to the certificate that contains the hash of the enclave
+    let extension_name = format!("{}.{}.{}.{}", VERACRUZ_RUNTIME_HASH_EXTENSION_ID[0], 
+                                                VERACRUZ_RUNTIME_HASH_EXTENSION_ID[1],
+                                                VERACRUZ_RUNTIME_HASH_EXTENSION_ID[2],
+                                                VERACRUZ_RUNTIME_HASH_EXTENSION_ID[3]);
+    let extension_value = format!("DER:{}", hex::encode(enclave_hash));
+    let custom_extension = openssl::x509::X509Extension::new(None, None, &extension_name, &extension_value)
+        .map_err(|err| {
+            println!("proxy-attestation-server::attestation::convert_csr_to_certificate X509Extension::new failed:{:?}", err);
+            err
+        })?;
+
+    cert_builder.append_extension(custom_extension)
+        .map_err(|err| {
+            println!("proxy-attestation-server::attestation::convert_csr_to_certificate append_extension for custom extension failed: {:?}", err);
+            err
+        })?;
+
+    let key_pem = {
+        let mut f = std::fs::File::open("../test-collateral/CAKey.pem")
```

Comment:
Can this path be pulled out as a top-level constant?

Reply:
Turns out, I already had the key value stored in a mutex, so this bit of code is quite unnecessary. Good catch.

Reply:
Should this be deleted, then?